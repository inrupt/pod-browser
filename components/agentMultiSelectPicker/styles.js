/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles } from "@solid/lit-prism-patterns";

const TABLE_MAX_HEIGHT = 460;
const MOBILE_TABLE_MIN_HEIGHT = 300;

export default function styles(theme) {
  return createStyles(theme, ["icons", "button", "table"], {
    "agent-multi-select-picker": {
      minHeight: MOBILE_TABLE_MIN_HEIGHT,
      display: "flex",
      flexDirection: "column",
      backgroundColor: theme.palette.common.white,
      borderRadius: theme.shape.borderRadius,
      border: `1px solid ${theme.palette.grey.A100}`,
      marginBottom: theme.spacing(2.4),
      height: "100%",
      maxHeight: TABLE_MAX_HEIGHT,
      overflowY: "auto",
      [theme.breakpoints.up("xs")]: {
        minHeight: "unset",
      },
    },
    "agent-multi-select-picker__tabs-and-button-container": {
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      borderBottom: `1px solid ${theme.palette.grey.A100}`,
    },
    "agent-multi-select-picker__agent-table": {
      width: `calc(100% - ${theme.spacing(1.6) * 2}px)`,
      boxSizing: "border-box",
      marginLeft: theme.spacing(1.6),
      marginRight: theme.spacing(1.6),
      "&& thead > tr": {
        minWidth: "100%",
        fontSize: theme.typography.caption.fontSize,
        alignItems: "center",
        borderBottom: `1px solid ${theme.palette.grey.A100}`,
        fontWeight: theme.typography.body1.fontWeight,
        color: theme.palette.text.primary,
      },
      "&& tbody tr": {
        borderBottom: `1px solid ${theme.palette.grey.A100}`,
      },
      "&& tbody td": {
        maxWidth: "100%",
      },
      "&& tbody :last-child": {
        borderBottom: "none",
      },
    },
    "agent-multi-select-picker__table-header": {
      display: "flex",
      margin: 0,
      padding: theme.spacing(0.5),
      textTransform: "capitalize",
    },
    "agent-multi-select-picker__tabs": {
      borderBottom: "none !important", // overriding the default border bottom so we can set it to the whole width of the container
    },
    "agent-multi-select-picker__empty": {
      display: "flex",
      flexDirection: "column",
      padding: theme.spacing(0.7, 1.4),
      textAlign: "center",
    },
    // hopefully we can remove these once we figure out how to test the Hidden component from MUI so we can use that instead
    "agent-multi-select-picker__mobile-only": {
      display: "flex",
      [theme.breakpoints.up("sm")]: {
        display: "none",
      },
    },
    "agent-multi-select-picker__desktop-only": {
      display: "none",
      [theme.breakpoints.up("sm")]: {
        display: "flex",
      },
    },
  });
}
