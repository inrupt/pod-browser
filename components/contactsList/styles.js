/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles, table } from "@solid/lit-prism-patterns";

const styles = (theme) => {
  const tableStyles = table.styles(theme);
  return createStyles(theme, ["container", "table", "icons"], {
    table: {
      "& tbody td": {
        "&:first-child": tableStyles["table__body-cell--width-preview"],
      },
    },
    "container-view": {
      marginTop: theme.spacing(1),
    },
    "container-view--menu-open": {
      [theme.breakpoints.up("sm")]: {
        paddingRight: "50%",
      },
      [theme.breakpoints.up("md")]: {
        paddingRight: "33.33%",
      },
      [theme.breakpoints.up("lg")]: {
        paddingRight: "25%",
      },
    },
    "container-menu": {
      display: "flex",
      justifyContent: "space-between",
      margin: theme.spacing(1, 0, 0),
    },
    avatar: {
      width: "30px",
      height: "30px",
    },
    addNewContact: {
      textAlign: "right",
    },
  });
};

export default styles;
