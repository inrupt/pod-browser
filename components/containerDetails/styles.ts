/* eslint-disable @typescript-eslint/no-explicit-any */

const styles: Record<string, any> = {
  centeredSection: {
    padding: "1rem",
  },
  raw: {
    height: "100%",
    width: "100%",
    maxHeight: "200px",
    overflow: "auto",
  },
  listItem: {
    display: "flex",
    justifyContent: "space-between",
  },
  detailText: {
    fontSize: "0.75rem",
  },
  typeValue: {
    marginLeft: "auto",
  },
};

export default styles;
