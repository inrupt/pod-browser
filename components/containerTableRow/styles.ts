/* eslint-disable @typescript-eslint/no-explicit-any */

const styles: Record<string, any> = {
  tableRow: {
    cursor: "pointer",
  },
  spinnerContainer: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "nowrap",
    justifyContent: "center",
  },
};

export default styles;
