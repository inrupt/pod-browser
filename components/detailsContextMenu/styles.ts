/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
import { PrismTheme } from "@solid/lit-prism-patterns";

const styles = (theme: PrismTheme) => ({
  "drawer__close-button": {
    width: 50,
    margin: "8px 0 0 -25px",
    backgroundColor: theme.palette.background.default,
    boxShadow: "1px 1px 3px #333",
    "&:hover": {
      backgroundColor: "#fff",
    },
  },
  "drawer__paper": {
    borderLeftStyle: "solid",
    borderLeftWidth: theme.spacing(1),
    borderLeftColor: theme.palette.grey["50"],
    width: "95%",
    [theme.breakpoints.up("sm")]: {
      width: "50%",
    },
    [theme.breakpoints.up("md")]: {
      width: "33.33%",
    },
    [theme.breakpoints.up("lg")]: {
      width: "25%",
    }
  },
});

export default styles;
