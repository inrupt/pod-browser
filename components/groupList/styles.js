/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

export default function styles(theme) {
  const lightPrimaryColor = "#F1EDFF"; // TODO: THIS MUST BE MOVED TO PRISM
  return {
    "group-list-header": {
      alignItems: "center",
      display: "flex",
      marginBottom: theme.spacing(1),
      padding: theme.spacing(1.6, 2.4, 1.2, 2.4),
    },
    "group-list-header__title": {
      ...theme.typography.h5,
      flexGrow: 1,
    },
    "group-list": {
      listStyle: "none",
      margin: 0,
      padding: 0,
    },
    "group-list__item": {},
    "group-list__link": {
      color: theme.palette.text.primary,
      display: "block",
      borderLeft: `solid 3px transparent`,
      padding: theme.spacing(1.6, 2.4, 1.6, 2.1),
      textDecoration: "none",
    },
    "group-list__link--selected": {
      background: lightPrimaryColor,
      borderColor: theme.palette.primary.dark,
      color: theme.palette.primary.main,
      fontWeight: theme.typography.fontWeightBold,
    },
    "group-list__icon": {
      fontSize: "0.8125rem",
    },
    "group-list__icon--selected": theme.icons.iconColor(
      theme.palette.primary.main
    ),
    "group-list__text": {
      paddingLeft: theme.spacing(1.4),
    },
  };
}
