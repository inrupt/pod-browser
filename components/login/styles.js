/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { createStyles } from "@solid/lit-prism-patterns";

export default function styles(theme) {
  return createStyles(theme, ["button"], {
    separator__wrap: {
      height: "2rem",
      margin: theme.spacing(2.4, 1.2),
      position: "relative",
      boxSizing: "border-box",
      marginBottom: theme.spacing(3.2),
    },
    "separator__centre-line": {
      textAlign: "center",
      position: "absolute",
      top: "50%",
      left: "50%",
      width: "100%",
      transform: "translate(-50%, -50%)",
      "&:before": {
        content: '""',
        position: "absolute",
        width: "100%",
        height: 1,
        top: "50%",
        left: 0,
        zIndex: -1,
        background: theme.palette.grey.A100,
      },
      "& span": {
        fontSize: "0.75rem",
        fontFamily: theme.typography.body1.fontFamily,
        color: theme.palette.grey[700],
        textTransform: "uppercase",
        backgroundColor: theme.palette.background.default,
        padding: "0.5rem 1rem",
        display: "inline-block",
      },
    },
    "provider-login-container": {
      display: "none",
      paddingTop: theme.spacing(1.6),
    },
    "provider-login-container--visible": {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
    },
    "info-button": {
      fontSize: theme.typography.button.fontSize,
    },
    "login-form": {
      [theme.breakpoints.up("sm")]: {
        maxWidth: 400,
        padding: theme.spacing(3.2),
      },
      textAlign: "center",
      fontStyle: "normal",
      marginBottom: theme.spacing(2.4),
      background: theme.palette.background.default,
      borderRadius: theme.shape.borderRadius,
      padding: theme.spacing(2.4),
      boxShadow: "0px 2px 4px 0px rgba(0,0,0,0.3)",
      "& h3": {
        fontFamily: theme.typography.h3.fontFamily,
        textTransform: "uppercase",
        fontSize: "0.75rem",
      },
    },
    "login-form__button": {
      width: "100%",
    },
    "login-form__logo": {
      marginBottom: theme.spacing(2.4),
    },
  });
}
