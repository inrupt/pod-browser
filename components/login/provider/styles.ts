import { StyleRules } from "@material-ui/styles";
import {
  PrismTheme,
  button,
} from "../../../lib/prism/node_modules/prism-patterns";

const styles = (theme: PrismTheme): StyleRules =>
  button.styles(theme) as StyleRules;

export default styles;
