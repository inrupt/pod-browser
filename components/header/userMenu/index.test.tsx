import * as ReactFns from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { ThemeProvider } from "@material-ui/core/styles";

import UserMenu from "./index";
import theme from "../../../src/theme";

describe("UserMenu", () => {
  test("renders a menu", () => {
    jest.spyOn(ReactFns, "useContext").mockImplementation(() => ({
      session: undefined,
    }));

    const tree = shallow(
      <ThemeProvider theme={theme}>
        <UserMenu />
      </ThemeProvider>
    );
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  // TODO: Write test for the various ways to toggle the user menu
  // TODO: Write test for checking profile image
});
