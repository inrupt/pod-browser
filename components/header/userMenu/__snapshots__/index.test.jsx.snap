// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu renders a menu 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-userMenu"
  >
    <div
      class="PodBrowser-header-banner__aside-menu PodBrowser-header-banner__aside-menu--popup"
    >
      <button
        aria-controls="UserMenu"
        aria-expanded="true"
        aria-haspopup="true"
        class="PodBrowser-header-banner__aside-menu-trigger"
        data-testid="user-menu-button"
        type="button"
      >
        <i
          aria-label="User Menu"
          class="PodBrowser-icon-user PodBrowser-header-banner__aside-menu-trigger-icon"
        />
      </button>
      <div
        aria-hidden="true"
        class="PodBrowser-header-banner__aside-menu-popup"
        id="UserMenu"
      >
        <ul
          class="PodBrowser-header-banner__user-menu"
        >
          <li
            class="PodBrowser-header-banner__user-menu-item"
            style="z-index: 100;"
          >
            <button
              class="PodBrowser-header-banner__user-menu-item-trigger"
              type="button"
            >
              <i
                class="PodBrowser-icon-user PodBrowser-header-banner__user-menu-item-icon"
              />
              <span
                class="PodBrowser-header-banner__user-menu-item-label"
              >
                Profile
              </span>
            </button>
          </li>
          <li
            class="PodBrowser-header-banner__user-menu-item"
            style="z-index: 100;"
          >
            <div
              role="button"
              tabindex="0"
            >
              <button
                class="PodBrowser-header-banner__user-menu-item-trigger"
                data-testid="logout-button"
                type="button"
              >
                <i
                  class="PodBrowser-icon-log-out PodBrowser-header-banner__user-menu-item-icon"
                />
                <span
                  class="PodBrowser-header-banner__user-menu-item-label"
                >
                  Log out
                </span>
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
